opm/simulators/wells/StandardWell_impl.hpp

```diff
@@ -279,36 +279,42 @@ namespace Opm
 
             // compute volume ratio between connection at standard conditions
             Value volumeRatio = bhp * 0.0; // initialize it with the correct type
-;
+
             if (FluidSystem::enableVaporizedWater() && FluidSystem::enableDissolvedGasInWater()) {
                 disOilVapWatVolumeRatio(volumeRatio, rvw, rsw, pressure,
                                         cmix_s, b_perfcells_dense, deferred_logger);
+                // only supported for gas-water case
+                assert(FluidSystem::phaseIsActive(FluidSystem::gasPhaseIdx));
+                assert(FluidSystem::phaseIsActive(FluidSystem::waterPhaseIdx));
+                assert(!FluidSystem::phaseIsActive(FluidSystem::oilPhaseIdx));
             } else {
+
                 if (FluidSystem::phaseIsActive(FluidSystem::waterPhaseIdx)) {
                     const unsigned waterCompIdx = Indices::canonicalToActiveComponentIndex(FluidSystem::waterCompIdx);
                     volumeRatio += cmix_s[waterCompIdx] / b_perfcells_dense[waterCompIdx];
                 }
+
+                if (FluidSystem::phaseIsActive(FluidSystem::oilPhaseIdx) && FluidSystem::phaseIsActive(FluidSystem::gasPhaseIdx)) {
+                    gasOilVolumeRatio(volumeRatio, rv, rs, pressure,
+                                    cmix_s, b_perfcells_dense, deferred_logger);
+                } else {
+                    if (FluidSystem::phaseIsActive(FluidSystem::oilPhaseIdx)) {
+                        const unsigned oilCompIdx = Indices::canonicalToActiveComponentIndex(FluidSystem::oilCompIdx);
+                        volumeRatio += cmix_s[oilCompIdx] / b_perfcells_dense[oilCompIdx];
+                    }
+                    if (FluidSystem::phaseIsActive(FluidSystem::gasPhaseIdx)) {
+                        if (! (FluidSystem::enableVaporizedWater() && FluidSystem::enableDissolvedGasInWater())) {
```

Comment:
`if (! (FluidSystem::enableVaporizedWater() && FluidSystem::enableDissolvedGasInWater())) ` is not needed here.